Formula/cascadia.rb

```diff
@@ -0,0 +1,21 @@
+class Cascadia < Formula
+  desc "Go cascadia package command-line CSS selector"
+  homepage "https://github.com/suntong/cascadia"
+  url "https://github.com/suntong/cascadia/archive/refs/tags/v1.2.6.tar.gz"
+  sha256 "eb7f88f23f176e74d1aeb3b34efeb7763de8ad5445566bc92ed9fee33e9cebd1"
+  license "MIT"
+  head "https://github.com/suntong/cascadia.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    # go mod init/tidy are needed until https://github.com/suntong/cascadia/issues/13 is fixed
+    system "go", "mod", "init", "github.com/suntong/cascadia"
+    system "go", "mod", "tidy"
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    assert_match "Version #{version}", shell_output("#{bin}/cascadia --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
> We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

I implemented this test based on other example formula I was looking at within the repo.

- https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula
  - > We want tests that don’t require any user input and test the basic functionality of the application. For example foo build-foo input.foo is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. **However, a bad test is better than no test at all.**

Reply:
Unfortunate that there isn't a `.html` fixture :(

- https://github.com/Homebrew/brew/tree/master/Library/Homebrew/test/support/fixtures

Reply:
```bash
⇒  echo "<foo><bar>aaa</bar><baz>bbb</baz></foo>" | cascadia --in --out --css 'foo > bar'
1 elements for 'foo > bar':
<bar>aaa</bar>
```

Reply:
you can download the test_resource, give other examples a look.

Reply:
Added in [`efadb02` (#120044)](https://github.com/Homebrew/homebrew-core/pull/120044/commits/efadb0238bf386c16e71d72b2a23f3a352a06730)

Reply:
give `brew style --fix cascadia` a try to fix the syntax issues.

Reply:
Done in [`41b3276` (#120044)](https://github.com/Homebrew/homebrew-core/pull/120044/commits/41b3276314ac78c2c85984939062068a4f888700)